instrumentation/cassandra/cassandra-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/cassandra/v4_0/CassandraNetAttributesExtractor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.cassandra.v4_0;
+
+import com.datastax.oss.driver.api.core.cql.ExecutionInfo;
+import com.datastax.oss.driver.api.core.metadata.Node;
+import io.opentelemetry.instrumentation.api.instrumenter.net.InetSocketAddressNetAttributesExtractor;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+final class CassandraNetAttributesExtractor
+    extends InetSocketAddressNetAttributesExtractor<CassandraRequest, ExecutionInfo> {
+
+  @Override
+  @Nullable
+  public String transport(CassandraRequest request) {
+    return null;
+  }
+
+  @Override
+  public @Nullable InetSocketAddress getAddress(
+      CassandraRequest request, @Nullable ExecutionInfo response) {
+    if (response == null) {
+      return null;
+    }
+    Node coordinator = response.getCoordinator();
+    if (coordinator == null) {
+      return null;
+    }
+    SocketAddress address = coordinator.getEndPoint().resolve();
```

Comment:
Wasn't happy to see the word resolve but looks like it's free...

https://github.com/datastax/java-driver/blob/4.x/core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultEndPoint.java#L39

Might be good to add a comment that most implementations of Endpoint should not incur a cost here

Reply:
ya, I had the same reaction when I saw `resolve`, comment added